src/lib/accessibility/Behaviors/Button/buttonBehavior.ts

```diff
@@ -3,14 +3,14 @@ import { Accessibility } from '../../interfaces'
 /**
  * @description
  *  Adds role='button' if element type is other than 'button'. This allows screen readers to handle the component as a button.
- *  Adds attribute 'aria-disabled=true' based on the property 'disabled'.
+ *  Adds attribute 'aria-disabled=true' based on the property 'disabled'. This can be overriden by directly providing 'aria-disabled' property to the component.
  */
 
 const buttonBehavior: Accessibility = (props: any) => ({
   attributes: {
     root: {
       role: props.as === 'button' ? undefined : 'button',
-      'aria-disabled': !!props['disabled'],
+      'aria-disabled': 'aria-disabled' in props ? props['aria-disabled'] : props['disabled'],
```

Comment:
why `in` check is necessary? seems that `props['aria-disabled'] || props['disabled']` would be sufficient enough, as this will make overrides only when `aria-disabled` prop is set

Reply:
no, actually the check is ok. our first option for `aria-disabled` is the manually specified `aria-disabled` on the component. If the component user does not specify it via props, we give `aria-disabled` the value of the `disabled` prop, which is always available.

The condition above will not work when, for instance you use `<Component aria-disabled="false" disabled="true">`. The `aria-disabled` value applied to the HTML element will be `true` in this case, which is not what we wanted.